xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-api/src/main/java/org/xwiki/extension/security/internal/ExtensionSecuritySolrClient.java

```diff
@@ -0,0 +1,163 @@
+@Singleton
+public class ExtensionSecuritySolrClient
+{
+    /**
+     * Mapping between the Live Data properties and the Solr properties.
+     */
+    private static final Map<String, String> LD_TO_SOLR = Map.of(
+        EXTENSION_ID, SOLR_FIELD_ID,
+        MAX_CVSS, SECURITY_MAX_CVSS,
+        FIX_VERSION, SECURITY_FIX_VERSION,
+        WIKIS, SECURITY_INSTALLED_WIKIS
+    );
+
+    @Inject
+    private ExtensionIndexStore extensionIndexStore;
+
+    @Inject
+    private SolrUtils solrUtils;
+
+    /**
+     * Perform a solr query returning the count of extensions with at least one known vulnerability.
+     *
+     * @return the query response of the query
+     * @throws IOException If there is a low-level I/O error
+     * @throws SolrServerException if there is an error on the server
+     */
+    public long getExtensionsCount() throws SolrServerException, IOException
+    {
+        SolrQuery solrQuery = new SolrQuery();
+        this.extensionIndexStore.createSolrQuery(new ExtensionQuery(), solrQuery);
+
+        initFilter(solrQuery);
+        QueryResponse search = this.extensionIndexStore.search(solrQuery);
+        return search.getResults().getNumFound();
+    }
+
+    /**
+     * Perform a solr query to obtain a list of extension with known vulnerabilities, according to a given live data
+     * query.
+     *
+     * @param liveDataQuery the live data query to use for the filters, sorts, and pagination
+     * @return the query response of the query
+     * @throws IOException If there is a low-level I/O error
+     * @throws SolrServerException if there is an error on the server
+     */
+    public QueryResponse solrQuery(LiveDataQuery liveDataQuery) throws SolrServerException, IOException
+    {
+        SolrQuery solrQuery = new SolrQuery();
+        this.extensionIndexStore.createSolrQuery(new ExtensionQuery(), solrQuery);
+        solrQuery.setRows(liveDataQuery.getLimit());
+        solrQuery.setStart(Math.toIntExact(liveDataQuery.getOffset()));
+
+        initSort(liveDataQuery, solrQuery);
+
+        initFilter(liveDataQuery, solrQuery);
+
+        return this.extensionIndexStore.search(solrQuery);
+    }
+
+    private void initFilter(LiveDataQuery liveDataQuery, SolrQuery solrQuery)
+    {
+        initFilter(solrQuery);
+        for (LiveDataQuery.Filter filter : liveDataQuery.getFilters()) {
+            String field = mapPropertyToField(filter.getProperty());
+            for (LiveDataQuery.Constraint constraint : filter.getConstraints()) {
+                String filterQueryString = this.solrUtils.toFilterQueryString(constraint.getValue());
+                if (StringUtils.isEmpty(filterQueryString)) {
+                    continue;
+                }
+                if (Objects.equals(field, SECURITY_MAX_CVSS)
+                    && Objects.equals(constraint.getOperator(), "contains"))
+                {
+                    solrQuery.addFilterQuery(String.format("%s:[%s TO 10]", field, filterQueryString));
```

Comment:
Not sure to understand: you already put that one in line 146 no? Also more generally this method might worth a bit of comment :)